pip/utils/packaging.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import absolute_import
+
+from email.parser import FeedParser
+import logging
+import sys
+
+from pip._vendor import pkg_resources
+from pip._vendor.packaging import specifiers
+from pip._vendor.packaging import version
+
+from pip import exceptions
+
+logger = logging.getLogger(__name__)
+
+
+def get_metadata(dist):
+    if (isinstance(dist, pkg_resources.DistInfoDistribution) and
+            dist.has_metadata('METADATA')):
+        return dist.get_metadata('METADATA')
+    elif dist.has_metadata('PKG-INFO'):
+        return dist.get_metadata('PKG-INFO')
+
+
+def check_requires_python(dist):
+    metadata = get_metadata(dist)
+    feed_parser = FeedParser()
+    feed_parser.feed(metadata)
+    pkg_info_dict = feed_parser.close()
+    requires_python = pkg_info_dict.get('Requires-Python')
+    if requires_python is None:
+        # The package provides no information
+        return
+    try:
+        requires_python_specifier = specifiers.SpecifierSet(requires_python)
+    except specifiers.InvalidSpecifier as e:
+        logger.debug(
+            "Package %s has an invalid Requires-Python entry %s - %s" % (
+                dist.project_name, requires_python, e))
+        return
```

Comment:
We used to ignore this information, so I'd say it is ok to keep ignoring it if we cannot parse it.
I'll bump the log level to warning to alert the user though.